File: aoc2024/day1a.py
import sys


def run():
    left = []
    right = []
    result = 0
    for line in sys.stdin:
        s = line.strip().split(" ")
        left.append(int(s[0]))
        right.append(int(s[-1]))

    left.sort()
    right.sort()

    for a, b in zip(left, right):
        result += abs(a - b)

    print(result)


if __name__ == "__main__":
    run()

File: src/aoc2024/day1a.rs
use std::io::BufRead;

pub fn run() {
    let lines: Vec<Vec<i32>> = std::io::stdin()
        .lock()
        .lines()
        .map(|x| {
            x.unwrap()
                .split_whitespace()
                .map(|y| y.parse::<i32>().unwrap())
                .collect()
        })
        .collect();
    let mut left: Vec<i32> = lines.iter().map(|x| *x.first().unwrap()).collect();
    let mut right: Vec<i32> = lines.iter().map(|x| *x.last().unwrap()).collect();

    left.sort();
    right.sort();

    let result: i32 = left
        .into_iter()
        .zip(right.into_iter())
        .map(|(a, b)| i32::abs(a - b))
        .sum();

    println!("{result}")
}
